### PR TITLE
#169005217 User signup backed by Postgres

### PR DESCRIPTION
**Pull request by joelatiam**
### [What does this PR do?](https://www.pivotaltracker.com/story/show/169005217)

Have the endpoint **/auth/signup** backed by  **Postgres RDBMS**

#### Description of Task to be completed?
After cloning this branch
have:
> latest version of postgres
> npm install
> all the packages in the package.json
> `npm start` : to run the app
> check the endpoint with **postman** `<host>/api/v2/signup` 
> test the endpoints by runing `npm test`

**N.B:** Referee to our [API documentation](https://teamwork-kg11.herokuapp.com/api/v1/docs/) to see the required request format and fields. This documentation is deprecated since it's linked with the Version one nevertheless both **v1** and **v2** send the same requests and expect same responses, the only difference is the **v1** and **v2** in the URL

#### [What are the relevant pivotal tracker stories?](https://www.pivotaltracker.com/story/show/169013200)
#168238817 #169013200

### CODE REVIEW

File: server/index.js [hunk 1]
```diff
@@ -20,6 +22,6 @@ app.get('/api/v1', (req, res) => {
 });
 
 
-app.listen(process.env.PORT, () => console.log(`Teamwork app listening on port ${process.env.PORT}!`));
+app.listen(port, () => console.log(`Teamwork app listening on port ${port}!`));
```
Comment (ekumamait) [inline server/index.js:25]: we do not merge `development tools` to `production code`

File: v2/server/config/config.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+const config = {
+  user: process.env.PGUSER,
+  host: process.env.PGHOST,
+  database: process.env.PGDATABASE,
+  password: process.env.PGPASSWORD || null,
+  port: process.env.PGPORT,
+};
+const pool = new pg.Pool(() => {
+  if (process.env.DATABASE_URL) {
```
Comment (ekumamait) [inline v2/server/config/config.js:13]: you can switch databases here

File: v2/server/controllers/articles.js [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+    }
+  } else {
+    errorMessage.missingToken(res);
+  }
+};
+
+const newArticle = (req, res) => {
+  const user = verifyToken(req, res);
+  if (user) {
+    // console.table(user);
```
Comment (ekumamait) [inline v2/server/controllers/articles.js:22]: kindly remove commented code

File: v2/server/doc.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
+import express from 'express';
+// import swaggerJsDoc from 'swagger-jsdoc';
+// import swaggerUi from 'swagger-ui-express';
+// import * as swaggerDoc from './document/swagger.json';
+import apiVersion from './helpers';
+
+const router = express.Router();
+
+// const swaggerOptions = {
```
Comment (ekumamait) [inline v2/server/doc.js:9]: Please remove all commented code from all your PRs

**Review by ekumamait — CHANGES_REQUESTED** [review]
Kindly make the necessary adjustments

File: v2/server/config/config.js (hunk 2 above)
Comment (joelatiam) [reply v2/server/config/config.js:13]: Thanks, I am currently working on it and have all the comments removed

File: server/index.js (hunk 1 above)
Comment (joelatiam) [reply server/index.js:25]: Thanks

**Comment by ekumamait** [comment]
Can you make sure to only return a token in the data object in the response data

**Comment by joelatiam** [comment]
> Can you make sure to only return a token in the data object in the response data

Do you mean that on a successful signup I should return only the token within the data object and ignore other user's details?

**Comment by ekumamait** [comment]
> > Can you make sure to only return a token in the data object in the response data
> 
> Do you mean that on a successful signup I should return only the token within the data object and ignore other user's details?

Yes please and you can do this for both `Auth` endpoints

**Comment by joelatiam** [comment]
> > > Can you make sure to only return a token in the data object in the response data
> > 
> > 
> > Do you mean that on a successful signup I should return only the token within the data object and ignore other user's details?
> 
> Yes please and you can do this for both `Auth` endpoints

Okay, noted.

**Pull request closed without merging**